lib/pager_duty/connection.rb

```diff
@@ -187,12 +187,13 @@ def initialize(token, token_type: :Token, debug: false)
     def get(path, request = {})
       # paginate anything being 'get'ed, because the offset/limit isn't intuitive
       request[:query_params] = {} if !request[:query_params]
-      page = (request[:query_params].delete(:page) || 1).to_i
-      limit = (request[:query_params].delete(:limit) || 100).to_i
+      page = request[:query_params].fetch(:page, 1).to_i
+      limit = request[:query_params].fetch(:limit, 100).to_i
       offset = (page - 1) * limit
-      request[:query_params] = request[:query_params].merge(offset: offset, limit: limit)
 
-      run_request(:get, path, **request)
+      query_params = request[:query_params].merge(offset: offset, limit: limit)
+
+      run_request(:get, path, **request.merge(query_params: query_params))
```

Comment:
Are there any side effects from the `request` hash having extra `page` and `limit` keys? I guess it is our method, so it can be fixed if it does.

Reply:
Apologies for missing this comment, and thanks for merging anyway 😓 This has been working for me with no adverse affects, but it's a good question to ask anyway.